src/components/FeatureViews/Scheduler/SessionView.js

```diff
@@ -71,54 +154,78 @@ class SessionView extends Component {
                     <Grid
                         className="session-view"
                         container
-                        ml={10}
-                        spacing={2}>
+                        spacing={2}
+                        direction={"row"}>
                         <Grid
                             item
-                            xs={12}>
+                            sm={12}
+                        >
                             <Typography
                                 align="left"
                                 className="session-view-title"
                                 variant="h3">
-                                {this.state.courseData.title}
+                                {this.state.courseData && this.state.courseData.title}
                             </Typography>
                         </Grid>
                         <Grid
                             align="left"
                             className="session-view-details"
                             container
-                            spacing={10}>
+                            spacing={16}
+                            xs={6}
+                        >
                             <Grid
                                 item
-                                lg={2}
-                                md={7}
                                 xs={6}>
                                 <Typography variant="h5"> Subject </Typography>
-                                <Typography varient="body1">{this.state.courseData.subject} </Typography>
+                                <Typography varient="body1">{this.state.courseData && this.state.courseData.subject} </Typography>
                             </Grid>
                             <Grid
                                 item
-                                xs="auto">
-                                <Typography variant="h5"> Date & Time </Typography>
-                                <Typography variant="body1">{this.state.sessionData.start}</Typography>
+                                xs={6}>
+                                <Typography variant="h5"> Room</Typography>
+                                <Typography variant="body1">{this.state.courseData && (this.state.courseData.room_id || "TBA")}</Typography>
```

Comment:
Parentheses may be misplaced; if `courseData` is null/undefined, the parenthesized OR is never reached, and nothing is shown.